src/data/repositories/TeamMemberD2Repository.ts

```diff
@@ -0,0 +1,46 @@
+import { D2Api, D2UserSchema } from "@eyeseetea/d2-api/2.36";
+import { TeamMember } from "../../domain/entities/incident-management-team/TeamMember";
+import { Id } from "../../domain/entities/Ref";
+import { TeamMemberRepository } from "../../domain/repositories/TeamMemberRepository";
+import { apiToFuture, FutureData } from "../api-futures";
+import { SelectedPick } from "@eyeseetea/d2-api/api";
+
+type D2User = SelectedPick<
+    D2UserSchema,
+    {
+        id: true;
+        name: true;
+        username: true;
+        email: true;
+        phoneNumber: true;
+    }
+>;
+export class TeamMemberD2Repository implements TeamMemberRepository {
+    constructor(private api: D2Api) {}
+
+    get(id: Id): FutureData<TeamMember> {
+        return apiToFuture(
+            this.api.metadata.get({
+                users: {
+                    fields: { id: true, name: true, email: true, phoneNumber: true },
+                    filter: { username: { eq: id } },
```

Comment:
Maybe for consistency add `username: true` to fields and to TeamMember because `incidentManagerName` is the username instead of the id. Also it's better to change to  `incidentManagerUsername` ?

Reply:
Added username: true , however do not understand the second part of the comment. Incident Manager is a type of role in the Team Member. Other Team Members could have other roles, so having the name incidentManagerUsername would not make sense, right?

Reply:
But I was thinking that `incidentManagerName` in `DiseaseOutbreakEventBaseAttrs` was the incident manager username selected in the form so I was populating this property with the username (tha's also why I need username in `TeamMember` entity, and also needed to populate the options in te incident manager selector)

Reply:
By the way, I tried to do it in the other PR but I had an error saying: `Property 'username' does not exist on type 'D2User'`

Reply:
ahh ok, i get it now.